### ISSUE TITLE
Multiple responses for one request, and their order

### ISSUE THREAD

**Issue by Ma233** [paritytech/litep2p#156]
Does the RequestResponse protocol allow the response side to send multiple responses? If yes, will the responses arrive at the request side in order?

**Comment by Ma233** [comment]
The connection closed after sending one response.

Fortunately, according to the libp2p specification (https://docs.libp2p.io/concepts/fundamentals/protocols/#binary-streams), the data is read in the same order as it was written.

A customized protocol can be achieved by referring to the custom_protocol example and the implementation of `RequestResponseProtocol`.

**Comment by dmitry-markin** [comment]
Yes, request-response protocol only allows sending one response per request. Basically, a new substream is opened for every request and closed after receiving the response.
